tgui/packages/tgui/interfaces/GeneralAlert.tsx

```diff
@@ -0,0 +1,105 @@
+import {
+  Button,
+  ByondUi,
+  Flex,
+  Image,
+  Section,
+  Stack,
+} from 'tgui-core/components';
+
+import { useBackend } from '../backend';
+import { Window } from '../layouts';
+import { capitalize } from './common/stringUtils';
+import { MinimapControllerData } from './MinimapController';
+
+export const GeneralAlert = () => {
+  const { data, act } = useBackend<MinimapControllerData>();
+  const {
+    title,
+    theme,
+    minimap_id,
+    minimap_markers,
+    placable_marker_states,
+    placable_marker_images,
+  } = data;
+
+  return (
+    <Window title={title} theme={theme} width={950} height={700}>
+      <Window.Content>
+        <Stack justify="center">
+          <Stack.Item>
+            <Section
+              title="Minimap"
+              fill
+              buttons={
+                <Button
+                  icon="undo"
+                  color="green"
+                  onClick={() => act('reset_scale')}
+                >
+                  Reset Map Scale
+                </Button>
+              }
+            >
+              <Flex>
+                <Flex.Item>
+                  <ByondUi
+                    params={{
+                      id: minimap_id,
+                      type: 'map',
+                    }}
+                    style={{
+                      width: '600px',
+                      height: '600px',
+                    }}
+                  />
+                </Flex.Item>
+              </Flex>
+            </Section>
+          </Stack.Item>
+          <Stack.Item grow>
+            <Section scrollable fill title="Alerts">
+              <Flex direction="column">
+                <Flex.Item>
+                  <Flex.Item>
+                    {Object.keys(minimap_markers).map((marker) => (
+                      <Stack key={data.minimap_markers[marker]}>
```

Comment:
Can you split these out into their own component? Nesting is radically deep

Reply:
Separated out to `MinimapIconMarker`